### ISSUE TITLE
Classification with CASTLE?

### ISSUE THREAD

**Issue by junzeliu** [trentkyono/CASTLE#1]
Hi,

Thank you for sharing the code. It was presented in the paper that CASTLE can also improve the performance of a classifier, and I'm trying to embed CASTLE as a regularizer for my classification task. May I ask how was the classification task accomplished with CASTLE? Should I simply replace the "supervised_loss" with a cross-entropy loss? In the current version of code, the target "y" is just one value. If I have a multi-class classification task, how should I modify the code to enable the CASTLE to output logits of many categories? Looking forward to your reply.

Thank you,
Junze

**Comment by trentkyono** [comment]
Hi Junze,

Correct, replace supervised_loss with cross-entropy.

Unfortunately we have not tried mult-class classification.  I suppose you
could treat it as a multi-task problem of binary variables, or a single
variable with multiple values (and use a regression-type loss).  Perhaps
there are other options as well.  Hope this works for you!

-Trent

**Comment by junzeliu** [comment]
I see. Thank you for your reply!

Best,
Junze

> Hi Junze, Correct, replace supervised_loss with cross-entropy. Unfortunately we have not tried mult-class classification. I suppose you could treat it as a multi-task problem of binary variables, or a single variable with multiple values (and use a regression-type loss). Perhaps there are other options as well. Hope this works for you!
> […](#)
> -Trent
